src/erasure-code/jerasure/ErasureCodeJerasure.h

```diff
@@ -32,28 +32,33 @@ class ErasureCodeJerasure : public ceph::ErasureCode {
   std::string rule_root;
   std::string rule_failure_domain;
   bool per_chunk_alignment;
+  uint64_t flags;
 
   explicit ErasureCodeJerasure(const char *_technique) :
-    k(0),
-    DEFAULT_K("2"),
-    m(0),
-    DEFAULT_M("1"),
-    w(0),
-    DEFAULT_W("8"),
-    technique(_technique),
-    per_chunk_alignment(false)
-  {}
+      k(0),
+      DEFAULT_K("2"),
+      m(0),
+      DEFAULT_M("1"),
+      w(0),
+      DEFAULT_W("8"),
+      technique(_technique),
+      per_chunk_alignment(false) {
+    flags = FLAG_EC_PLUGIN_PARTIAL_READ_OPTIMIZATION |
+      FLAG_EC_PLUGIN_PARTIAL_WRITE_OPTIMIZATION |
+      FLAG_EC_PLUGIN_ZERO_INPUT_ZERO_OUTPUT_OPTIMIZATION |
+      FLAG_EC_PLUGIN_PARITY_DELTA_OPTIMIZATION;
+
+    if (0 == strcmp(technique, "reed_sol_van")) {
```

Comment:
How about
```suggestion
    if ("reed_sol_van"sv == technique) {
```
here and in other places as well?

Reply:
The string_view operator seems to be causing issue at this location. I will investigate and fix if possible.

Reply:
Fixed.